ci/shellcheck.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
+
+success=
+for file in $(git ls-files | grep '\.sh$'); do
+  if ! shellcheck "${file}"; then
+    success=1
+  fi
+done
+
+if [ -n "${success}" ]; then
+  exit 1
+fi
```

Comment:
I feel like the `success` variable here reflects... the opposite of success?

How about something like:
```sh
exitstatus=0
for ...
  if ! shellcheck ... exitstatus=1
done

exit $exitstatus
```

That's not super great either but mostly because exit status has the opposite sense of what I usually expect, which is normal. Also, I'm assuming that this script is exec'd, not sourced; otherwise, unconditionally exiting would probably be bad.

Reply:
That's fair. What about renaming it to `failed`?

Reply:
Sure, I'll take it.